### ISSUE TITLE
NullPointerException while parsing

### ISSUE THREAD

**Issue by tkvangorder** [openrewrite/rewrite#1825]
This was encountered while working through an example with a design partner.

```
org.openrewrite.java.JavaParsingException: Failed to convert for the following cursor stack:--- BEGIN PATH ---
JCCompilationUnit(sourceFile = ******.java)
--- END PATH ---

        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1470)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1476)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convertAll(ReloadableJava8ParserVisitor.java:1513)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitCompilationUnit(ReloadableJava8ParserVisitor.java:506)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitCompilationUnit(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:550)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8Parser.lambda$parseInputs$3(ReloadableJava8Parser.java:204)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at org.openrewrite.java.ReloadableJava8Parser.parseInputs(ReloadableJava8Parser.java:226)
        at org.openrewrite.java.Java8Parser.parseInputs(Java8Parser.java:44)
        at org.openrewrite.Parser.parse(Parser.java:41)
        at org.openrewrite.maven.MavenMojoProjectParser.processTestSources(MavenMojoProjectParser.java:243)
        at org.openrewrite.maven.MavenMojoProjectParser.listSourceFiles(MavenMojoProjectParser.java:128)
        at org.openrewrite.maven.AbstractRewriteMojo.listResults(AbstractRewriteMojo.java:182)
        at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:46)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.NullPointerException
        at com.sun.tools.javac.code.Symbol$MethodSymbol.params(Symbol.java:1572)
        at com.sun.tools.javac.code.Symbol$MethodSymbol.getParameters(Symbol.java:1632)
        at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodArgumentSignature(ReloadableJava8TypeSignatureBuilder.java:229)
        at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodSignature(ReloadableJava8TypeSignatureBuilder.java:220)
        at org.openrewrite.java.ReloadableJava8TypeMapping.methodDeclarationType(ReloadableJava8TypeMapping.java:482)
        at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:212)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
        at org.openrewrite.java.ReloadableJava8TypeMapping.variableType(ReloadableJava8TypeMapping.java:362)
        at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:204)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:287)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:279)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:836)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1903)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1447)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitImport(ReloadableJava8ParserVisitor.java:695)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitImport(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCImport.accept(JCTree.java:579)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1447)

java.lang.NullPointerException
        at com.sun.tools.javac.code.Symbol$MethodSymbol.params(Symbol.java:1572)
        at com.sun.tools.javac.code.Symbol$MethodSymbol.getParameters(Symbol.java:1632)
        at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodArgumentSignature(ReloadableJava8TypeSignatureBuilder.java:229)
        at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodSignature(ReloadableJava8TypeSignatureBuilder.java:220)
        at org.openrewrite.java.ReloadableJava8TypeMapping.methodDeclarationType(ReloadableJava8TypeMapping.java:482)
        at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:212)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
        at org.openrewrite.java.ReloadableJava8TypeMapping.variableType(ReloadableJava8TypeMapping.java:362)
        at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:204)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:287)
        at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:279)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:836)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1903)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1447)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitImport(ReloadableJava8ParserVisitor.java:695)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitImport(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCImport.accept(JCTree.java:579)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1447)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1476)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.convertAll(ReloadableJava8ParserVisitor.java:1513)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitCompilationUnit(ReloadableJava8ParserVisitor.java:506)
        at org.openrewrite.java.ReloadableJava8ParserVisitor.visitCompilationUnit(ReloadableJava8ParserVisitor.java:66)
        at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:550)
        at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
        at org.openrewrite.java.ReloadableJava8Parser.lambda$parseInputs$3(ReloadableJava8Parser.java:204)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at org.openrewrite.java.ReloadableJava8Parser.parseInputs(ReloadableJava8Parser.java:226)
        at org.openrewrite.java.Java8Parser.parseInputs(Java8Parser.java:44)
        at org.openrewrite.Parser.parse(Parser.java:41)
        at org.openrewrite.maven.MavenMojoProjectParser.processTestSources(MavenMojoProjectParser.java:243)
        at org.openrewrite.maven.MavenMojoProjectParser.listSourceFiles(MavenMojoProjectParser.java:128)
        at org.openrewrite.maven.AbstractRewriteMojo.listResults(AbstractRewriteMojo.java:182)
        at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:46)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

**Comment by traceyyoshima** [comment]
This looks related to an issue we've seen before: https://github.com/openrewrite/rewrite/issues/1377

**Comment by traceyyoshima** [comment]
The potential cause of the NPE is subject to change, since TypeMapping may be reworked. Unassigning myself to focus on other issues.

**Comment by traceyyoshima** [comment]
I attempted to reproduce the NPE for a different design partner in #1377 and failed to do so.
I looked at the NPE again, since I understand the inner working of the compiler better now, but still can't reproduce the issue.

The stack trace from the original issue occurs on the same line numbers in `com.sun.tools.javac.code.Symbol$MethodSymbol.params(Symbol.java:1572)` with a previous line of `com.sun.tools.javac.code.Symbol$MethodSymbol.getParameters(Symbol.java:1632)`.

Stack trace:
```java
18:58:42 Caused by: java.lang.NullPointerException
18:58:42 	at com.sun.tools.javac.code.Symbol$MethodSymbol.params(Symbol.java:1572)
18:58:42 	at com.sun.tools.javac.code.Symbol$MethodSymbol.getParameters(Symbol.java:1632)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodArgumentSignature(ReloadableJava8TypeSignatureBuilder.java:229)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeSignatureBuilder.methodSignature(ReloadableJava8TypeSignatureBuilder.java:220)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.methodDeclarationType(ReloadableJava8TypeMapping.java:476)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:214)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.variableType(ReloadableJava8TypeMapping.java:356)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.classType(ReloadableJava8TypeMapping.java:205)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:60)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:281)
18:58:42 	at org.openrewrite.java.ReloadableJava8TypeMapping.type(ReloadableJava8TypeMapping.java:273)
18:58:42 	at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:827)
18:58:42 	at org.openrewrite.java.ReloadableJava8ParserVisitor.visitMemberSelect(ReloadableJava8ParserVisitor.java:64)
18:58:42 	at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1903)
18:58:42 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
18:58:42 	at org.openrewrite.java.ReloadableJava8ParserVisitor.convert(ReloadableJava8ParserVisitor.java:1438)
```

**Comment by tkvangorder** [comment]
This bug report into netbeans looks very similar to what we are seeing: 

https://bz.apache.org/netbeans/show_bug.cgi?id=256636

**Comment by traceyyoshima** [comment]
> This bug report into netbeans looks very similar to what we are seeing:
> 
> https://bz.apache.org/netbeans/show_bug.cgi?id=256636

I had already checked the bugs I could find when the source code was provided, but the NPE did not occur.

I just swept through each of the NPE reports in jdk11u-dev after the last commit in java-8. I attempted to use the tests as sources to reproduce the NPE without any exceptions. I also checked changes to `Symbol` and looked for references throughout commits to `getParameterTypes()`, but didn't find any relevant changes.

I think we'll need an example in code that reproduces the issue.
